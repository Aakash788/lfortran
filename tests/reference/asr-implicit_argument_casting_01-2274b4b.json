{
    "basename": "asr-implicit_argument_casting_01-2274b4b",
    "cmd": "lfortran --show-asr --implicit-argument-casting --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/implicit_argument_casting_01.f90",
    "infile_hash": "7592210ad6046c12c005301338a547d35c4c130ce9d5dd46c8afe8f6",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-implicit_argument_casting_01-2274b4b.stdout",
    "stdout_hash": "b5db79ddb1e3f565f748b8ea7d297dec5eecc5ac022bf930973387cc",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}