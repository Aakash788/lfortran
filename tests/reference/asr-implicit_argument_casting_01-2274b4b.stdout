(TranslationUnit
    (SymbolTable
        1
        {
            idz_realcomp:
                (Function
                    (SymbolTable
                        2
                        {
                            a:
                                (Variable
                                    2
                                    a
                                    [n]
                                    Unspecified
                                    ()
                                    ()
                                    Default
                                    (Array
                                        (Real 8)
                                        [((IntegerConstant 1 (Integer 4))
                                        (Var 2 n))]
                                        PointerToDataArray
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            n:
                                (Variable
                                    2
                                    n
                                    []
                                    Unspecified
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    idz_realcomp
                    (FunctionType
                        [(Integer 4)
                        (Array
                            (Real 8)
                            [((IntegerConstant 1 (Integer 4))
                            (FunctionParam
                                0
                                (Integer 4)
                                ()
                            ))]
                            PointerToDataArray
                        )]
                        ()
                        Source
                        Implementation
                        ()
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                    )
                    []
                    [(Var 2 n)
                    (Var 2 a)]
                    [(=
                        (Var 2 a)
                        (RealConstant
                            12.500000
                            (Real 8)
                        )
                        ()
                    )
                    (Return)]
                    ()
                    Public
                    .false.
                    .false.
                    ()
                ),
            idzp_svd:
                (Function
                    (SymbolTable
                        3
                        {
                            c_f_pointer:
                                (ExternalSymbol
                                    3
                                    c_f_pointer
                                    5 c_f_pointer
                                    lfortran_intrinsic_iso_c_binding
                                    []
                                    c_f_pointer
                                    Public
                                ),
                            c_loc:
                                (ExternalSymbol
                                    3
                                    c_loc
                                    5 c_loc
                                    lfortran_intrinsic_iso_c_binding
                                    []
                                    c_loc
                                    Public
                                ),
                            eps:
                                (Variable
                                    3
                                    eps
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Real 8)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            i:
                                (Variable
                                    3
                                    i
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            isi:
                                (Variable
                                    3
                                    isi
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            ls:
                                (Variable
                                    3
                                    ls
                                    []
                                    Unspecified
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            real:
                                (ExternalSymbol
                                    3
                                    real
                                    13 real
                                    lfortran_intrinsic_builtin
                                    []
                                    real
                                    Private
                                ),
                            w:
                                (Variable
                                    3
                                    w
                                    []
                                    Unspecified
                                    ()
                                    ()
                                    Default
                                    (Array
                                        (Complex 8)
                                        [(()
                                        ())]
                                        DescriptorArray
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            w_cast:
                                (Variable
                                    3
                                    w_cast
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Pointer
                                        (Array
                                            (Real 8)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    idzp_svd
                    (FunctionType
                        [(Integer 4)
                        (Array
                            (Complex 8)
                            [(()
                            ())]
                            DescriptorArray
                        )]
                        ()
                        Source
                        Implementation
                        ()
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                    )
                    [idz_realcomp]
                    [(Var 3 ls)
                    (Var 3 w)]
                    [(=
                        (Var 3 isi)
                        (IntegerConstant 1 (Integer 4))
                        ()
                    )
                    (=
                        (Var 3 ls)
                        (IntegerConstant 5 (Integer 4))
                        ()
                    )
                    (CPtrToPointer
                        (PointerToCPtr
                            (GetPointer
                                (ArrayItem
                                    (Var 3 w)
                                    [(()
                                    (Var 3 isi)
                                    ())]
                                    (Complex 8)
                                    ColMajor
                                    ()
                                )
                                (Pointer
                                    (Complex 8)
                                )
                                ()
                            )
                            (CPtr)
                            ()
                        )
                        (Var 3 w_cast)
                        (ArrayConstant
                            [(IntegerConstant 1000 (Integer 4))]
                            (Array
                                (Integer 4)
                                [((IntegerConstant 1 (Integer 4))
                                (IntegerConstant 1 (Integer 4)))]
                                FixedSizeArray
                            )
                            ColMajor
                        )
                        ()
                    )
                    (SubroutineCall
                        1 idz_realcomp
                        ()
                        [((Var 3 ls))
                        ((ArrayPhysicalCast
                            (Var 3 w_cast)
                            DescriptorArray
                            PointerToDataArray
                            (Pointer
                                (Array
                                    (Real 8)
                                    [(()
                                    ())]
                                    PointerToDataArray
                                )
                            )
                            ()
                        ))]
                        ()
                    )
                    (Print
                        ()
                        [(ArrayItem
                            (Var 3 w)
                            [(()
                            (Var 3 isi)
                            ())]
                            (Complex 8)
                            ColMajor
                            ()
                        )]
                        ()
                        ()
                    )
                    (If
                        (RealCompare
                            (IntrinsicFunction
                                Abs
                                [(RealBinOp
                                    (Cast
                                        (ArrayItem
                                            (Var 3 w)
                                            [(()
                                            (Var 3 isi)
                                            ())]
                                            (Complex 8)
                                            ColMajor
                                            ()
                                        )
                                        ComplexToReal
                                        (Real 4)
                                        ()
                                    )
                                    Sub
                                    (Cast
                                        (RealConstant
                                            12.500000
                                            (Real 8)
                                        )
                                        RealToReal
                                        (Real 4)
                                        (RealConstant
                                            12.500000
                                            (Real 4)
                                        )
                                    )
                                    (Real 4)
                                    ()
                                )]
                                0
                                (Real 4)
                                ()
                            )
                            Gt
                            (RealConstant
                                0.000000
                                (Real 4)
                            )
                            (Logical 4)
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (Return)]
                    ()
                    Public
                    .false.
                    .false.
                    ()
                ),
            iso_c_binding:
                (IntrinsicModule lfortran_intrinsic_iso_c_binding),
            lfortran_intrinsic_builtin:
                (IntrinsicModule lfortran_intrinsic_builtin),
            main:
                (Program
                    (SymbolTable
                        10
                        {
                            idzp_svd:
                                (Function
                                    (SymbolTable
                                        11
                                        {
                                            c_f_pointer:
                                                (ExternalSymbol
                                                    11
                                                    c_f_pointer
                                                    5 c_f_pointer
                                                    lfortran_intrinsic_iso_c_binding
                                                    []
                                                    c_f_pointer
                                                    Public
                                                ),
                                            c_loc:
                                                (ExternalSymbol
                                                    11
                                                    c_loc
                                                    5 c_loc
                                                    lfortran_intrinsic_iso_c_binding
                                                    []
                                                    c_loc
                                                    Public
                                                ),
                                            ls:
                                                (Variable
                                                    11
                                                    ls
                                                    []
                                                    Unspecified
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            w:
                                                (Variable
                                                    11
                                                    w
                                                    []
                                                    Unspecified
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Complex 8)
                                                        [(()
                                                        ())]
                                                        DescriptorArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    idzp_svd
                                    (FunctionType
                                        [(Integer 4)
                                        (Array
                                            (Complex 8)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )]
                                        ()
                                        Source
                                        Interface
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                    )
                                    []
                                    [(Var 11 ls)
                                    (Var 11 w)]
                                    []
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            w:
                                (Variable
                                    10
                                    w
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Array
                                        (Complex 8)
                                        [((IntegerConstant 1 (Integer 4))
                                        (IntegerConstant 5 (Integer 4)))]
                                        FixedSizeArray
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    main
                    [iso_c_binding]
                    [(SubroutineCall
                        10 idzp_svd
                        ()
                        [((IntegerConstant 5 (Integer 4)))
                        ((ArrayPhysicalCast
                            (Var 10 w)
                            FixedSizeArray
                            DescriptorArray
                            (Array
                                (Complex 8)
                                [((IntegerConstant 1 (Integer 4))
                                (IntegerConstant 5 (Integer 4)))]
                                DescriptorArray
                            )
                            ()
                        ))]
                        ()
                    )]
                )
        })
    []
)
